CommandLine.c

```diff
@@ -350,21 +358,28 @@ int CommandLine_run(const char* name, int argc, char** argv) {
    CRT_init(settings, flags.allowUnicode);
 
    MainPanel* panel = MainPanel_new();
-   ProcessList_setPanel(pl, (Panel*) panel);
+   MainPanel* genericDataPanel = MainPanel_new();
 
    MainPanel_updateLabels(panel, settings->ss->treeView, flags.commFilter);
 
    State state = {
       .settings = settings,
       .ut = ut,
       .pl = pl,
-      .mainPanel = panel,
+      .mainPanel = panel, // comment this line to break signal panel ( press 'k' button )
       .header = header,
-      .pauseProcessUpdate = false,
-      .hideProcessSelection = false,
+      .pauseUpdate = false,
+      .hideSelection = false,
    };
 
-   MainPanel_setState(panel, &state);
+   //MainPanel_setState(panel, &state);
+   panel->state = &state;
+   genericDataPanel->state = &state;
```

Comment:
Should this be ... ?
```
   MainPanel_setState(panel, &state);
   MainPanel_setState(genericDataPanel, &state);
```

Reply:
@natoscott I think we need a new panel working in parallel with/instead MainPanel. This is a big part of what I'm talking about so far...
this panel will be minimal and doesn't have `Nice (f7)` or `Kill (f9)` panels [[dito](https://github.com/htop-dev/htop/blob/main/MainPanel.c#L25)]
This will fix a lot of issues, maybe one of them is the issue mentioned above `--enable-debug`, not sure, this part needs surgery ^^" 
but MainPanel has `Class(Process)`: https://github.com/htop-dev/htop/blob/bc22bee6505a4953d20eed96899d929e769a7baa/MainPanel.c#L215

also, @BenBE told me before how to do that: @natoscott I need your help here...
> BenBe: I think it might be worth looking into a direction where the current Panel code is just used as a derived class ProcessListPanel (showing a process list), while most of its code is split into a base class GenericPanel (doing mostly nothing). From that GenericPanel class you could then split another derived class "FancyPanel" for doing all the Fancy stuff ;-)
> 
> Sohaib: Could you explain this more please?
> 
> BenBe: Basically what I was going at was not adopting the current Panel class to also show generic stuff, but to use the existing inheritance stuff so you can just push different kinds of Panels ("Process Lists" vs. "Generic Panels") into the list of tabs handled by the ScreenManager code.
> The way to do this mostly boils down to split the current Panel implementation into the basic "draw the panel" code (extracted from current source BasePanel class) and the code drawing the process list (a new ProcessListPanel class).
> The ScreenManager then instead of listing plain Panels would list BasePanels, which could be ProcessListPanels (working like they do now) or GenricStuffPanels (doing something generic).

Reply:
@BenBE any chance you can help @smalinux here with some sample code changes to get him started on this aspect?  thanks!

Reply:
Will have to wait for next week; not in town tomorrow and the day after (giving a talk for a small conference). Changes-wise it boils down to creating inherited classes similar to what can be seen with `ProcessList` and `<Platform>ProcessList` with the Panel managing the `ProcessList` stuff getting most of the current process list draw code. Extra points when rendering the header/list items is split in a way so that for the generic tabs and the process list the actual rendering code is the same, but only creating/filling the content buffers differs. As said, I can elaborate next week …